spec/unit/ace/transport_app_spec.rb

```diff
@@ -83,6 +83,16 @@ def app
     expect(last_response.status).to eq(200)
   end
 
+  describe '#trusted_facts' do
+    it {
+      expect(described_class.trusted_facts('foo.domain.com')).to eq(authenticated: "remote",
+                                                                    certname: "foo.domain.com",
+                                                                    domain: "domain.com",
+                                                                    extensions: {},
+                                                                    hostname: "foo")
+    }
+  end
```

Comment:
please also add a testcase for when the certname does not contain a dot, to make sure that the parsing code is not confused by that.